### PR TITLE
[Enhance] adjust buffer size of LocalExchange

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Current buffer size of LocalExchange is one chunk per dop, it has some drawbacks:
- Cause more schedule overhead: the sink pipeline must be scheduled after put one chunk, same as source
- Could not balance the speed between source and sink

So, here we change the buffer to 8 chunks. The experiment result should significant improvement through this change.

## Experiment result
- The chunk size is 4096
- Test some local-exchange intensive query

| Query | Before | After | 
| --- | ---- | ---- |
| `SELECT COUNT(c1),COUNT(c2),COUNT(c3) FROM ( SELECT LO_SUPPKEY c1, LO_ORDERDATE c2, sum(LO_REVENUE) OVER( PARTITION BY LO_SUPPKEY ORDER BY LO_ORDERDATE ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) c3 FROM ssb_100g.lineorder_flat ) as a;` |  29227ms | 6913ms | 
| `SELECT COUNT(c1),count(c2),count(c3) FROM ( SELECT LO_CUSTKEY c1, LO_ORDERDATE c2, sum(LO_REVENUE) OVER( PARTITION BY LO_CUSTKEY ORDER BY LO_ORDERDATE ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) c3 FROM ssb_100g.lineorder_flat ) as a;`| 25486ms | 6928ms|

### CODE REVIEW

**Review by liuyehcf — DISMISSED** [review]

**Comment by silverbullet233** [comment]
could you provide some performance result?

File: be/src/exec/pipeline/pipeline_builder.cpp [hunk 1]
```diff
@@ -14,7 +14,8 @@ OpFactories PipelineBuilderContext::maybe_interpolate_local_broadcast_exchange(R
     }
 
     auto pseudo_plan_node_id = next_pseudo_plan_node_id();
-    auto mem_mgr = std::make_shared<LocalExchangeMemoryManager>(state->chunk_size() * num_receivers * num_receivers);
+    auto mem_mgr = std::make_shared<LocalExchangeMemoryManager>(state->chunk_size() * num_receivers * num_receivers *
```
Comment (kangkaisen) [inline be/src/exec/pipeline/pipeline_builder.cpp:17]: why double multiply num_receivers？

Comment (mofeiatwork) [reply be/src/exec/pipeline/pipeline_builder.cpp]: That's wrong, the correct formula should be `chunk_size * num_receiver * 8`.

**Review by silverbullet233 — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

**Pull request merged**